tests/ext-popup.test.js

```diff
@@ -0,0 +1,48 @@
+import Popup from '../src/lib/ext-popup';
+import * as ExtListen from '../src/lib/ext-listen';
+
+test('test init popup when no extension is being monitored', async () => {
```

Comment:
This test seems to be testing the popup when the monitor is active, and so if I'm not wrong the description of the test should be changed accordingly.

Reply:
@atiqueahmedziad I forgot to ask you this in one of the other review comments: which are the other scenarios that you think you should test on the popup?

Writing down a short list in text form (just as short test descriptions like the one passed to `test("...", async () {...})`) is helpful to plan the tests to write (and would help me to give you feedback on the list of test scenario even before you have turned all the tests in the list into code).

Feel free to reply to this comment with the list of test cases you are thinking of and I'm going to take a look to it asap.

Reply:
I want to add the following tests - 

`test("test popup UI when extensions are not being monitored", async () { ... })`
`test("test popup UI when error is thrown while initialize monitoring", async () { ... })`
`test("test popup UI when error is thrown while stop monitoring", async () { ... })`
`test("viewExtPage() method is called upon clicking on 'view activity logs page' button", async () {...}))`

Reply:
maybe `"test popup UI when extensions are not being monitored"` could be expanded a bit to test a more complete scenario, e.g. how about:
- `"starting monitoring all extension from the popup"`
  - popup loaded initially in the "not monitoring" state (while we mock the internal helpers to let us simulate this state)
  - check that the UI is in the expected state
  - click the start monitor (after mocking the internal helpers that would let us know that the monitor did start successfully)
  - check that the UI has been updated as expected

how that sounds to you?

Reply:
Nit, I would also tweak a bit the description of these two:

- "viewExtPage() method is called upon clicking on 'view activity logs page' button" => "clicking 'view activity logs page' should open the activitylog extension page"
- "test popup UI when an error is thrown while initializing monitoring and stop monitoring" => "errors on start/stopping monitoring should be shown in the popup UI"

(just to express them in terms of the scenario being tested, instead of using a description that is focused on the particular code tested)

Reply:
Sounds good. :-) 
Let me expand them in the following - 
- "stop monitoring all the extensions from popup"
    - popup loaded initially in the monitoring state.
    - check that the UI is in the expected state
    - click the stop monitor button
    - check the UI has been updated as expected.
- "error is thrown while initializing monitoring and stop monitoring"
    - popup loaded initially with no error message
    - click on start monitor button/ stop monitor button
    - an error is thrown
    - UI should be updated as expected i.e. the error message is showed.
- "clicking 'view activity logs page' should open the activitylog extension page"
    - popup initially loaded
    - click on "view activity logs page" button
    - it should open a new tab with extension page